nemo/collections/asr/metrics/wer.py

```diff
@@ -146,48 +186,462 @@ def ctc_decoder_predictions_tensor(
+            # after forcing partial text conversion of the token.
+            if token != token_text:
+                # If there are any partially or fully built sub-word token ids, construct to text.
+                # Note: This is "old" subword, that occurs *after* current sub-word has started.
+                if len(built_token) > 0:
+                    word_offsets.append(
+                        {
+                            "word": decode_tokens_to_str(built_token),
+                            "start_offset": offsets[i]["start_offset"],
+                            "end_offset": offsets[i]["end_offset"],
+                        }
+                    )
+
+                # Prepare list of new sub-word ids
+                built_token.clear()
+                built_token.append(char)
+            else:
+                # If the token does not contain any sub-word start mark, then the sub-word has not completed yet
+                # Append to current sub-word list.
+                built_token.append(char)
+
+        # Inject the start offset of the first token to word offsets
+        # This is because we always skip the delay the injection of the first sub-word due to the loop
+        # condition and check whether built token is ready or not.
+        # Therefore without this forced injection, the start_offset appears as off by 1.
+        word_offsets[0]["start_offset"] = offsets[0]["start_offset"]
+
+        # If there are any remaining tokens left, inject them all into the final word offset.
+        # Note: The start offset of this token is the start time of the first token inside build_token.
+        # Note: The end offset of this token is the end time of the last token inside build_token
+        if len(built_token) > 0:
+            word_offsets.append(
+                {
+                    "word": decode_tokens_to_str(built_token),
+                    "start_offset": offsets[-(len(built_token))]["start_offset"],
+                    "end_offset": offsets[-1]["end_offset"],
+                }
+            )
+        built_token.clear()
+
+        return word_offsets
+
+    @property
+    def preserve_alignments(self):
+        return self._preserve_alignments
+
+    @preserve_alignments.setter
+    def preserve_alignments(self, value):
+        self._preserve_alignments = value
+
+        if hasattr(self, 'decoding'):
+            self.decoding.preserve_alignments = value
+
+    @property
+    def compute_timestamps(self):
+        return self._compute_timestamps
+
+    @compute_timestamps.setter
+    def compute_timestamps(self, value):
+        self._compute_timestamps = value
+
+        if hasattr(self, 'decoding'):
+            self.decoding.compute_timestamps = value
+
+
+class CTCCharDecoding(AbstractCTCDecoding):
+    """
+    Used for performing CTC auto-regressive / non-auto-regressive decoding of the logprobs.
```

Comment:
It has similar description to CTCDecoding.

Reply:
No need to repeat it if they are the same.

Reply:
Done, good point. Lets keep it consistent with RNNTDecoding